### DISCUSSION TITLE
Combine `dagger listen` and `dagger session`

### DISCUSSION THREAD

**Discussion by sipsma** [dagger/dagger#12441] [maintainers]
These two commands are quite similar to the point where they should probably be combined into one. This will require some thinking since `dagger session` has a json output consumed the SDKs that isn't ideal for humans using `dagger listen` among other little issues like that. These can most likely be addressed through flags that tweak the behavior (i.e. a flag to enable json output that the sdks use).

This is a non-essential spin off task from the previous overall effort to update the session interface: https://github.com/dagger/dagger/issues/3830

**Comment by TomChv** [comment]
Is this issue still relevant @sipsma ? And is there any work I can do on that to help?

**Comment by sipsma** [comment]
@TomChv we should probably do this someday, but I wouldn't consider it especially high priority at the moment. It's a small simplification to just internal code.
